test/contrib/batch_test.py

```diff
@@ -162,6 +163,46 @@ def test_wait_on_job_failed(self):
             self.bc.wait_on_job(job_id)
             self.assertTrue('log line 1' in context.exception)
 
+    def test_wait_on_job_failed_before_start(self):
```

Comment:
According to travis, this test case doesn't cover the change. Could you please take a look? Thanks.